internal/index/indexscanner/scanner.go

```diff
@@ -86,38 +86,59 @@ func LoadPluginByName(pluginsDir, pluginName string) (index.Plugin, error) {
 // ReadPluginFromFile loads a file from the FS. When plugin file not found, it
 // returns an error that can be checked with os.IsNotExist.
 func ReadPluginFromFile(path string) (index.Plugin, error) {
-	f, err := os.Open(path)
-	if os.IsNotExist(err) {
-		// TODO(ahmetb): we should use go1.13+ errors.Is construct at call sites to evaluate if an error is os.IsNotExist
-		return index.Plugin{}, err
-	} else if err != nil {
-		return index.Plugin{}, errors.Wrap(err, "failed to open index file")
+	var plugin index.Plugin
+	err := readFromFile(path, &plugin)
+	if err != nil {
+		return plugin, err
 	}
-	return ReadPlugin(f)
+	return plugin, errors.Wrap(validation.ValidatePlugin(plugin.Name, plugin), "plugin manifest validation error")
```

Comment:
Oh no.. looks like we swapped the duplication of boilerplate code for opening the file with duplication of our business logic. IMHO it's better if this function continues to call `ReadPlugin` and we duplicate the file-open boilerplate.

This also means, that `readFromFile` becomes obsolete and should be inlined in `ReadReceiptFromFile`.

@ahmetb what do you think?

Reply:
It seems ok to me now.

Reply:
Ok, then let's settle with the current version.